### PR TITLE
Small refactoring

### PR DESCRIPTION
**Pull request by Wadeck**
Main issue was to retrieve the User by username potentially.

- bump plugin parent pom
- also add escape-by-default into Jelly files
- add final to a static variable

@reviewbybees @jtnord (especially for the getById without creation)

### CODE REVIEW

**Comment by daniel-beck** [comment]
Isn't it a feature that users can log in to Jenkins with their display name?

This seems to disable that.

**Comment by Wadeck** [comment]
As mentioned @daniel-beck we could break a feature that is potentially used : the LDAP search using the FullName (against cn ?).

The other changes are still interesting, but nothing really important.

**Comment by Wadeck** [comment]
@reviewbybees done

File: src/main/java/org/jenkinsci/plugins/reverse_proxy_auth/ReverseProxySecurityRealm.java [hunk 1]
```diff
@@ -117,7 +117,7 @@
 	 * See http://msdn.microsoft.com/en-us/library/aa746475(VS.85).aspx for the syntax by example.
 	 * WANTED: The specification of the syntax.
 	 */
-	public static String GROUP_SEARCH = System.getProperty(LDAPSecurityRealm.class.getName()+".groupSearch",
+	public static final String GROUP_SEARCH = System.getProperty(LDAPSecurityRealm.class.getName()+".groupSearch",
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/reverse_proxy_auth/ReverseProxySecurityRealm.java]: After this change it won't be possible to change them dynamically in System Groovy Scripts

**Comment by oleg-nenashev** [comment]
FindBugs will likely grumble, we will fix it in the master branch

**Pull request merged**